code/modules/mob/living/simple_animal/abnormality/he/wayward_passenger.dm

```diff
@@ -0,0 +1,264 @@
+		)
+	return pick(result_text_list[level])
+
+//*** Breach mechanics ***
+
+/mob/living/simple_animal/hostile/abnormality/wayward/BreachEffect(mob/living/carbon/human/user)
+	icon_state = "wayward_breach"
+	playsound(src, 'sound/abnormalities/thunderbird/tbird_zombify.ogg', 45, FALSE, 5)//this is the sound effect used for Tomerry in the lovetown reception
+	..()
+
+//*** Teleport code ***//
+/mob/living/simple_animal/hostile/abnormality/wayward/proc/TryTeleport(turf/teleport_target)//argument is for the playable version, currently unused
+	if(teleport_cooldown > world.time)
+		return FALSE
+	teleport_cooldown = world.time + teleport_cooldown_time//so it doesn't get called twice by life()
+	icon_state = "wayward_tpstart"
+	playsound(src, 'sound/abnormalities/wayward_passenger/teleport.ogg', 100, 1)
+	var/list/teleport_potential = list()//TODO: skip this process entirely for the playable version
+	for(var/mob/living/L in GLOB.mob_living_list)
+		if(L.stat == DEAD || L.z != z || L.status_flags & GODMODE || faction_check_mob(L))
+			continue
+		if(!ishuman(L))
+			continue
+		if(ishuman(L))
+			var/mob/living/carbon/human/H = L
+			if(H.is_working)
+				continue
+		teleport_potential += get_turf(L)
+	if(!LAZYLEN(teleport_potential))
+		if(!LAZYLEN(GLOB.department_centers))
+			return
+		var/turf/P = pick(GLOB.department_centers)
+		teleport_potential += P
+	can_act = FALSE
+	LoseTarget()
+	if(!teleport_target)
+		teleport_target = pick(teleport_potential)
+	SLEEP_CHECK_DEATH(4)
+	playsound(src, 'sound/abnormalities/wayward_passenger/teleport2.ogg', 100, 1)
+	density = FALSE
+	var/obj/effect/portal/abno_warp/P1 = new(get_turf(src))
+	forceMove(teleport_target)
+	var/obj/effect/portal/abno_warp/P2 = new(teleport_target)
+	P1.link_portal(P2)
+	P2.link_portal(P1)
+	icon_state = "wayward_tpend"
+	playsound(src, 'sound/abnormalities/wayward_passenger/teleport2.ogg', 100, 1)
+	SLEEP_CHECK_DEATH(2 SECONDS) //2 seconds to teleport
+	density = TRUE
+	SLEEP_CHECK_DEATH(4)
+	can_act = TRUE
+	icon_state = "wayward_breach"
+	can_dash = TRUE
+
+/*** Dash code ***/
+/mob/living/simple_animal/hostile/abnormality/wayward/proc/Dash(target)
+	if(charging || dash_cooldown > world.time)
+		return
+	can_act = FALSE
+	can_dash = FALSE
+	update_icon()//TODO: dash sprite
+	dash_cooldown = world.time + dash_cooldown_time
+	charging = TRUE
+	var/dir_to_target = get_dir(get_turf(src), get_turf(target))
+	been_hit = list()
+	addtimer(CALLBACK(src, .proc/Do_Dash, dir_to_target, 0), 3 SECONDS)//how long it takes for the dash to initiate.
+	playsound(src, 'sound/abnormalities/wayward_passenger/attack1.ogg', 150, 1)
+
+/mob/living/simple_animal/hostile/abnormality/wayward/proc/Do_Dash(move_dir, times_ran)
+	can_act = TRUE
+	var/stop_charge = FALSE
+	if(times_ran >= dash_num)
+		stop_charge = TRUE
+		icon_state = "wayward_breach"
+	var/turf/T = get_step(get_turf(src), move_dir)
+	if(!T)
+		charging = FALSE
+		return
```

Comment:
```suggestion
	if(times_ran >= dash_num)
		stop_charge = TRUE
	var/turf/T = get_step(get_turf(src), move_dir)
	if(!T)
		charging = FALSE
		icon_state = "wayward_breach"
		return
```
First one doesn't need it, because stop_charge being TRUE already sets the icon to that, the second one may need it in case of an emergency stop.